### PR TITLE
[MM-48618, MM-48639] Fix thread replies not clearing text box and not syncing across sessions

### PR DESCRIPTION
**Pull request by mylonsuren**
#### Summary

- Fixes an issue where sending a message in a comment reply by pressing `Enter` would result in the text box failing to clear. When a message is sent the text box unfocused which automatically triggered a draft save, and so this "new" draft would appear in the text box. This PR adds a flag to only save drafts when not in a submitting state.

- Also fixes an issue where drafts would not update for a thread until you closed and reopened the RHS.

- Changes text on drafts page to use "Any" instead of "All".

#### Ticket Link

- https://mattermost.atlassian.net/browse/MM-48618
- https://mattermost.atlassian.net/browse/MM-48639

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by amyblais** [comment]
@EricssonLiu @koox00 Please help prioritize this PR review.

**Review by koox00 — APPROVED** [review]

**Review by EricssonLiu — APPROVED** [review]
LGTM 🛖

**Comment by jgilliam17** [comment]
Thanks @mylonsuren 
Seeing few issues when starting a draft and when adding mention to draft on the RHS 
Screenshot 2022-11-28 161238
Screenshot 2022-11-28 161208

**Comment by mylonsuren** [comment]
> Thanks @mylonsuren Seeing few issues when starting a draft and when adding mention to draft on the RHS Screenshot 2022-11-28 161238 Screenshot 2022-11-28 161208

Thanks @jgilliam17, that is a known issue with draft upserts which will be fixed in https://github.com/mattermost/mattermost-server/pull/21752. We can hold off on this PR until that gets merged.

**Comment by jgilliam17** [comment]
Sounds good - please mention me when ready for retest - thanks 🙂

**Comment by mylonsuren** [comment]
> Sounds good - please mention me when ready for retest - thanks 🙂

@jgilliam17 Could you test the first ticket though if possible? https://mattermost.atlassian.net/browse/MM-48618 should be fixed regardless of if the draft is saved or not. The second ticket https://mattermost.atlassian.net/browse/MM-48639 can wait.

**Comment by jgilliam17** [comment]
Also [e2e report](https://mattermost-cypress-report.s3.amazonaws.com/267940-ed7b499-onprem-ent-webapp-pr-11654-mm-ee-test:ed7b499/mochawesome.html), I forgot to post the link earlier. I'll re-run after server PR is merged.
I'll check the other ticket now.

**Comment by jgilliam17** [comment]
verified [MM-48618](https://mattermost.atlassian.net/browse/MM-48618) is fixed - RHS textbox is cleared after posting

**Comment by jgilliam17** [comment]
Verified help text and header info updated to  both show "Any messages you've started will show here"

**Comment by mylonsuren** [comment]
@jgilliam17 Sorry about that, just pushed a fix for it, looks like there was an issue with the caret positioning when the suggestion list came up.

**Comment by jgilliam17** [comment]
First [E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/268686-b98bf6f-onprem-ent-webapp-pr-11654-mm-ee-test:b98bf6f/mochawesome.html)- feature flag for Global Drafts true
Few drafts related failures differ from master:
MM-T211
MM-T4640
MM-T4641
MM-T453_2
(MM-T2371 - disregard, fails on other PRs)

**Comment by jgilliam17** [comment]
@mylonsuren 
Verified drafts are updated across sessions, including drafts in threads. Also, reply threads are clearing after posting.
Noticed one issue - Initially I used the PR test server without GlobalDrafts flag set. Once I switched flag to true, all existing drafts on the test server were lost.

**Comment by jgilliam17** [comment]
E2E report (feature flag false) is taking longer than expected. I'll check in the morning and post the results here.

**Comment by jgilliam17** [comment]
e2e cycle failed to complete, updating and restarting the cycle

**Comment by jgilliam17** [comment]
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/268919-0a15631-onprem-ent-webapp-pr-11654-mm-ee-test:0a15631/mochawesome.html) - looks good, 4 test failures differ from master but don't appear related to PR changes
Filed for follow up: https://mattermost.atlassian.net/browse/MM-48745

**Comment by jgilliam17** [comment]
Additional follow up tickets:
https://mattermost.atlassian.net/browse/MM-48747
https://mattermost.atlassian.net/browse/MM-48748

**Review by jgilliam17 — APPROVED** [review]
Thanks @mylonsuren 
Tested, looks good to merge.
Few follow up issues filed and linked previously.

**Pull request merged**
